Adding some explanations and removing spelling errors (#253)
nf-core/differentialabundance: docs/output.md

  - `qc/`: Directory containing quality control plots from initial processing e.g. DESeq2
    - `*.png`
  - `exploratory/`: Directory containing standalone plots from exploratory analysis. Plots are stored in directories named for the main coloring variable used.
    - `[coloring variable]/png/boxplot.png`: Boxplot visualization of abundance distributions
    - `[coloring variable]/png/density.png`: Density visualization of abundance distributions
    - `[coloring variable]/png/pca2d.png`: 2-dimensional PCA plot
    - `[coloring variable]/png/pca3d.png`: 3-dimensional PCA plot
    - `[coloring variable]/png/sample_dendrogram.png`: A sample clustering dendrogram
    - `[coloring variable]/png/mad_correlation.png`: Outlier prediction plots using median absolute deviation (MAD)
  - `differential/`: Directory containing standalone plots from differential analysis. Plots are stored in directories named for the associated contrast.

>     - `[contrast]/png/volcano.png`: Volcano plots of -log(10) p value against log(2) fold changes. Each dot in the plot represents one gene. The higher genes in the volcano plot are highly significant, and the genes located in right or left side of the plot are highly dysregulated.

Reviewer: Sorry, the output.md is not a great place for extensive explanation of plots. Also, 'volcano plot' has special meaning most people understand.

Reviewer 2: sorry, I disagree here. https://nf-co.re/eager/2.5.1/docs/output is for me a prime example of pipeline documentation including thorough explanations of the plots. (even though I agree that vulcano plots are quite established, a bit of explanation doesn't hurt imo)

    - `[coloring variable]/png/sample_dendrogram.png`: A sample clustering dendrogram
    - `[coloring variable]/png/mad_correlation.png`: Outlier prediction plots using median absolute deviation (MAD)
  - `differential/`: Directory containing standalone plots from differential analysis. Plots are stored in directories named for the associated contrast.

>     - `[contrast]/png/volcano.png`: Volcano plots of -log(10) p value against log(2) fold changes. Each dot in the plot represents one gene. The higher genes in the volcano plot are highly significant, and the genes located in right or left side of the plot are highly dysregulated.

Reviewer 3: @pinin4fjords where would you rather have this?

>     - `[contrast]/png/volcano.png`: Volcano plots of -log(10) p value against log(2) fold changes. Each dot in the plot represents one gene. The higher genes in the volcano plot are highly significant, and the genes located in right or left side of the plot are highly dysregulated.

Reviewer: I don't think the additional description adds anything. The statements added follow from what was already there (if you understand what a fold change and -log10(p value) are. If this level of detail is added here we'd have to add it for all files and I think it would just make things wordier without adding additional meaning.

<summary>Output files</summary>

- `tables/`

  - `annotation1/`: Directory containing annotation matrices generated in the course of analysis
    - `[array platform].annotation.tsv`: Annotations derived from an array platform
    - `[GTF name].anno.tsv`: Species wise annotations derived from a GTF in RNA-seq analysis
  - `processed_abundance/`: Directory containing processed abundance values from initial processing from e.g. DESeq2 or Affy:

>     - `[contrast_name].normalised_counts.tsv`: Normalized counts table (DESeq2). Rows are ensembl IDs and columns are samples.

Reviewer 4: ```suggestion
    - `[contrast_name].normalised_counts.tsv`: Normalized counts table (DESeq2). Rows are Ensembl IDs and columns are samples.
```

Uppercase seems to be correct (from a quick [web search](http://www.ensembl.org/Help/Faq?id=488)).

    - `[contrast_name].vst.tsv`: Normalised counts table with a variance-stabilising transform (DESeq2)
    - `raw.matrix.tsv`: RMA background corrected matrix (Affy)
    - `normalised.matrix.tsv`: RMA background corrected and normalised intensities matrix (Affy)
  - `differential/`: Directory containing tables of differential statistics reported by differential modules such as DESeq2

    - `[contrast_name].deseq2.results.tsv`: Results of DESeq2 differential analysis (RNA-seq)

>     - `OR [contrast_name].limma.results.tsv`: Results of Limma differential analysis (Affymetrix arrays)
> 
>   - `gsea/`: Directory containing tables of differential gene set analysis from GSEA (where enabled)
> 
>     - `[contrast_name].[deseq2|limma].results.tsv`: Results of DESeq2 differential analyis (RNA-seq) OR Limma differential analysis (Affymetrix arrays, GEO studies, Maxquant proteomics studies)
>     - `[contrast_name].[deseq2|limma].results_filtered.tsv`: Results of DESeq2 differential analyis (RNA-seq) OR Limma differential analysis (Affymetrix arrays, GEO studies, Maxquant proteomics studies); filtered for differentially abundant entries

Reviewer 4: ```suggestion
    - `OR [contrast_name].limma.results.tsv`: Results of limma differential analysis (Affymetrix arrays)

  - `gsea/`: Directory containing tables of differential gene set analysis from GSEA (where enabled)

    - `[contrast_name].[deseq2|limma].results.tsv`: Results of DESeq2 differential analyis (RNA-seq) OR limma differential analysis (Affymetrix arrays, GEO studies, Maxquant proteomics studies)
    - `[contrast_name].[deseq2|limma].results_filtered.tsv`: Results of DESeq2 differential analyis (RNA-seq) OR limma differential analysis (Affymetrix arrays, GEO studies, Maxquant proteomics studies); filtered for differentially abundant entries
```

The _limma_ authors published the tool in lowercase (see [publication](https://academic.oup.com/nar/article/43/7/e47/2414268)).

    - `[array platform].annotation.tsv`: Annotations derived from an array platform
    - `[GTF name].anno.tsv`: Species wise annotations derived from a GTF in RNA-seq analysis
  - `processed_abundance/`: Directory containing processed abundance values from initial processing from e.g. DESeq2 or Affy:
    - `[contrast_name].normalised_counts.tsv`: Normalized counts table (DESeq2). Rows are ensembl IDs and columns are samples.
    - `[contrast_name].vst.tsv`: Normalized counts table with a variance-stabilizing transform (DESeq2)
    - `raw.matrix.tsv`: RMA background corrected matrix (Affy)
    - `normalised.matrix.tsv`: RMA background corrected and normalized intensities matrix (Affy)
  - `differential/`: Directory containing tables of differential statistics reported by differential modules such as DESeq2

    - `[contrast_name].deseq2.results.tsv`: Results of DESeq2 differential analysis (RNA-seq)

>     - `OR [contrast_name].limma.results.tsv`: Results of Limma differential analysis (Affymetrix arrays) including ensembl ID, log2FC, and adjusted P. Value

Reviewer: There's a lot of conditional logic in the workflow that causes certain files to appear in certain circumstances. We don't need to express that conditionality here.

    - `raw.matrix.tsv`: RMA background corrected matrix (Affy)
    - `normalised.matrix.tsv`: RMA background corrected and normalised intensities matrix (Affy)
  - `differential/`: Directory containing tables of differential statistics reported by differential modules such as DESeq2
    - `[contrast_name].[deseq2|limma].results.tsv`: Results of DESeq2 differential analyis (RNA-seq) OR Limma differential analysis (Affymetrix arrays, GEO studies, Maxquant proteomics studies)

>     - `[contrast_name].[deseq2|limma].results_filtered.tsv`: Results of DESeq2 differential analyis (RNA-seq) OR Limma differential analysis (Affymetrix arrays, GEO studies, Maxquant proteomics studies); filtered for differentially abundant entries

Reviewer 5: This change deletes quite a few changes that were done on the dev branch but not yet merged to master (the results_filtered files should be documented and Maxquant should be mentioned in both lines); please ensure that your branch is up-to-date with dev

  - `gsea/`: Directory containing tables of differential gene set analyis from GSEA (where enabled)
    - `[contrast_name].deseq2.results.tsv`: Results of DESeq2 differential analysis (RNA-seq)
    - `OR [contrast_name].limma.results.tsv`: Results of Limma differential analysis (Affymetrix arrays)
  - `gsea/`: Directory containing tables of differential gene set analysis from GSEA (where enabled)
    - `[contrast]/[contrast].gsea_report_for_[condition].tsv`: A GSEA report table for each side of each contrast
  - `gprofiler2/`: Directory containing tables of differential gene set analyis from gprofiler2 (where enabled)
    - `[contrast]/[contrast].gprofiler2.all_enriched_pathways.tsv`: A gprofiler2 report table for all enrichment results

>     - `[contrast]/[contrast].gprofiler2.[source].sub_enriched_pathways.tsv`: A gprofiler2 report table of enriched pathways from one specific source/database, e.g. REAC

Reviewer 5: here as well, this should not be deleted

**Suggested course of action:** Examine the Principal Component Analysis (PCA) and metadata association plots generated by the workflow. Identify variables associated with components that contribute significantly to the variance in your data. Include these variables as covariates in the contrasts table's blocking column to account for their effects on gene expression. By incorporating these unaccounted sources of variance into your analysis, you improve the accuracy and power to detect differential expression.

#### 5. Biological complexity and pathway-level effects:

**Problem:** The experimental intervention may not lead to observable differential expression at the individual gene level, but there may be coordinated changes at the pathway or functional level.

**Suggested course of action:** Utilise pathway analysis tools such as Gene Set Enrichment Analysis (GSEA), available in this workflow. These tools evaluate the enrichment of gene sets or functional annotations to identify broader biological processes influenced by the experimental intervention.

> In this analysis, the differentially expressed genes (DEGs) divides into different groups, based on their effect of specific signaling pathways. In each group, genes will be sorted and assigned by scores based on their logFC and significance level. It is a good approach to find hub DEGs in disease-related signaling pathways.

Reviewer 4: ```suggestion
In this analysis, the differentially expressed genes (DEGs) are divided into different groups, based on their effect of specific signaling pathways. In each group, genes will be sorted and assigned by scores based on their logFC and significance level. It is a good approach to find hub DEGs in disease-related signaling pathways.
```
